hudi-common/src/main/java/org/apache/hudi/common/util/CompactionUtils.java

```diff
@@ -159,9 +160,30 @@ private static List<Pair<HoodieInstant, HoodieCompactionPlan>> getCompactionPlan
       HoodieTableMetaClient metaClient, Function<HoodieTableMetaClient, HoodieTimeline> filteredTimelineSupplier,
       Function<String, HoodieInstant> requestedInstantWrapper) {
     List<HoodieInstant> filteredInstants = filteredTimelineSupplier.apply(metaClient).getInstants();
+
     return filteredInstants.stream()
-        .map(instant -> Pair.of(instant, getCompactionPlan(metaClient, requestedInstantWrapper.apply(instant.getTimestamp()))))
-        .collect(Collectors.toList());
+          .map(instant ->  {
+            HoodieCompactionPlan compactionPlan = null;
+            try {
+              compactionPlan = getCompactionPlan(metaClient, requestedInstantWrapper.apply(instant.getTimestamp()));
+            } catch (HoodieException e) {
+              if (e.getCause() instanceof IOException && e.getCause().getMessage().contains("Not an Avro data file")) {
+                // try reloading metaclient so that compaction plan could have been fully serialized.
+                try {
+                  Thread.sleep(5);
```

Comment:
We should never expect to sleep in catch block. And the problem may exist for all the plan files, not just compaction.